pycode/memilio-surrogatemodel/memilio/surrogatemodel/ode_secir_simple/model.py

```diff
@@ -0,0 +1,190 @@
+#############################################################################
+# Copyright (C) 2020-2022 German Aerospace Center (DLR-SC)
+#
+# Authors: Agatha Schmidt, Henrik Zunker
+#
+# Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#############################################################################
+import numpy as np
+import pandas as pd
+import os
+import pickle
+import tensorflow as tf
+import matplotlib.pyplot as plt
+from memilio.surrogatemodel.ode_secir_simple import data_generation
+from memilio.surrogatemodel.ode_secir_simple import different_networks
+
+
+def plotCol(
+        inputs, labels, model=None, plot_col='InfectedSymptoms',
+        max_subplots=8):
+    """! Plot prediction of the model and label for one compartment.
+
+    @param inputs test inputs for model prediction. 
+    @param labels test labels. 
+    @param model trained model. 
+    @param plot_col string name of compartment to be plotted. 
+    @param max_subplots number of plot to be plotted. 
+    """
+
+    input_width = inputs.shape[1]
+    label_width = labels.shape[1]
+
+    plt.figure(figsize=(12, 8))
+    cols = np.array([
+        'Susceptible', 'Exposed', 'InfectedNoSymptoms', 'InfectedSymptoms',
+        'InfectedSevere', 'InfectedCritical', 'Recovered', 'Dead'])
+    plot_col_index = np.where(cols == plot_col)[0][0]
+    max_n = min(max_subplots, inputs.shape[0])
+
+    for n in range(max_n):
+        plt.subplot(max_n, 1, n+1)
+        plt.ylabel(f'{plot_col}')
+
+        input_array = inputs[n].numpy()
+        label_array = labels[n].numpy()
+        plt.plot(np.arange(0, input_width), input_array[:, plot_col_index],
+                 label='Inputs', marker='.', zorder=-10)
+        plt.scatter(
+            np.arange(input_width, input_width + label_width),
+            label_array[:, plot_col_index],
+            edgecolors='k', label='Labels', c='#2ca02c', s=64)
+
+        if model is not None:
+            input_series = tf.expand_dims(inputs[n], axis=0)
+            pred = model(input_series)
+            pred = pred.numpy()
+            plt.scatter(np.arange(input_width, input_width+pred.shape[-2]),
```

Comment:
pred.shape[-2] always equals len(label_array), doesnt it?

Reply:
pred.shape[-2] (or pred.shape[1]) is the number time steps we want to predict.
But yes, it would also be possible to get this value from the label dataset.